cmd/paramgen/main.go

```diff
@@ -0,0 +1,152 @@
+// Copyright © 2022 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	"go/format"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/conduitio/conduit-connector-sdk/cmd/paramgen/internal"
+)
+
+type Args struct {
+	output     string
+	structName string
+	path       string
+}
+
+func main() {
+	log.SetFlags(0)
+	log.SetPrefix("paramgen: ")
+
+	// parse the command arguments
+	args := parseArgs()
+
+	// parse the sdk parameters
+	params, err := internal.ParseParameters(args.path, args.structName)
+	if err != nil {
+		log.Fatalf("paramgen failed to parse parameters: %v\n", err)
+	}
+
+	// convert the parameter map to a string
+	str := paramsToString(params)
+
+	processTemplate(str, args)
+}
+
+const (
+	outputDirName = "/paramgen"
+	paramGenTmpl  = `// Code generated by ParamGen. DO NOT EDIT.
+// Source: github.com/conduitio/conduit-connector-sdk/cmd/paramgen
+
+// Package paramgen is a generated ParamGen package.
+package paramgen
+
+import sdk "github.com/conduitio/conduit-connector-sdk"
+
+func getParameters() map[string]sdk.Parameter {
+  return {{.}}
```

Comment:
Gofmt would not be happy with the two spaces, a tab should be used for indentation.

More importantly - the way we use the template right now doesn't bring us a lot of value. We are essentially just inserting a string into this template, this could easily be done with `fmt.Sprintf`. What I expected is that we would write a template that operates on `map[string]sdk.Parameter` and knows how to print that map as Go code. This would give us more control over the code that we output.